redisinsight/ui/src/components/init/AppInit.tsx

```diff
@@ -1,16 +1,40 @@
-import React, { ReactElement, useEffect } from 'react'
+import React, { ReactElement, useCallback, useEffect } from 'react'
 import { useDispatch, useSelector } from 'react-redux'
-import { appInitSelector, initializeAppAction, STATUS_SUCCESS } from 'uiSrc/slices/app/init'
+import {
+  appInitSelector,
+  initializeAppAction,
+  STATUS_FAIL,
+  STATUS_INITIAL,
+  STATUS_SUCCESS,
+} from 'uiSrc/slices/app/init'
+import ConnectivityError from 'uiSrc/components/connectivity-error/ConnectivityError'
 import PagePlaceholder from '../page-placeholder'
 
-const AppInit = ({ children }: { children: ReactElement }) => {
+type Props = {
+  children: ReactElement
+  onSuccess?: () => void,
+  onFail?: () => void,
+}
+
+const AppInit = ({ children, onSuccess, onFail }: Props) => {
   const dispatch = useDispatch()
-  const { status } = useSelector(appInitSelector)
+  const {
+    status,
+    error = 'Something went wrong, please try again later',
+  } = useSelector(appInitSelector)
+
+  const initApp = useCallback(() => dispatch(initializeAppAction(onSuccess, onFail)), [])
 
   useEffect(() => {
-    dispatch(initializeAppAction())
+    if (status === STATUS_INITIAL) {
```

Comment:
I had to add this because of the tests, they would go into a loop without it. Not sure why

Reply:
If I remove this, the error I'm getting is about a Redux action:
![image](https://github.com/user-attachments/assets/f28e4b4d-59aa-46e7-ad72-ab45f87097f3)

It's complaining about this dispatch, because something isn't mocked or setup correctly for the test:
`dispatch(initializeAppAction(onSuccess, onFail))`

I think we're still missing some critical tests for this component. We are testing the component in different states with different props, but there are no tests that actually test the interaction of the component with the Redux store. I still think we need to remove this `if` condition, and add more tests that use mock service worker to handle API calls, and use a mocked store so the component can actually dispatch the actions and re-render to the correct states based on those interactions. Otherwise, we don't have any test coverage for how this component behaves in the real world when connected to Redux and actually makes API requests. What are your thoughts?